kubevirt_api/tests/test_unit.py

```diff
@@ -0,0 +1,253 @@
+        "vmi_uid:f1f3ae4b-f81f-406f-a574-f12e7e3ba4f2",
+        "vmi_phase:Running",
+    ]
+    aggregator.assert_metric(
+        "kubevirt_api.vmi.count",
+        value=1,
+        tags=vmi_tags,
+    )
+
+    aggregator.assert_all_metrics_covered()
+    aggregator.assert_metrics_using_metadata(get_metadata_metrics())
+
+
+def test_check_sends_zero_count_for_vms(dd_run_check, aggregator, instance, mocker):
+    mocker.patch("requests.get", wraps=mock_http_responses)
+
+    check = KubeVirtApiCheck("kubevirt_api", {}, [instance])
+
+    check._setup_kube_client = lambda: None
+    check.kube_client = MagicMock()
+    check.kube_client.get_vms.return_value = []
+    check.kube_client.get_vmis.return_value = []
+
+    dd_run_check(check)
+
+    aggregator.assert_metric(
+        "kubevirt_api.can_connect",
+        value=1,
+        tags=healthz_tags,
+    )
+
+    aggregator.assert_metric(
+        "kubevirt_api.vm.count",
+        value=0,
+        tags=[
+            "kube_cluster_name:test-cluster",
+        ],
+    )
+
+
+def test_check_sends_zero_count_for_vmis(dd_run_check, aggregator, instance, mocker):
+    mocker.patch("requests.get", wraps=mock_http_responses)
+
+    check = KubeVirtApiCheck("kubevirt_api", {}, [instance])
+
+    check._setup_kube_client = lambda: None
+    check.kube_client = MagicMock()
+    check.kube_client.get_vms.return_value = []
+    check.kube_client.get_vmis.return_value = []
+
+    dd_run_check(check)
+
+    aggregator.assert_metric(
+        "kubevirt_api.can_connect",
+        value=1,
+        tags=healthz_tags,
+    )
+
+    aggregator.assert_metric(
+        "kubevirt_api.vmi.count",
+        value=0,
+        tags=[
+            "kube_cluster_name:test-cluster",
+        ],
+    )
+
+
+def test_emits_zero_can_connect_when_service_is_down(dd_run_check, aggregator, instance):
+    check = KubeVirtApiCheck("kubevirt_api", {}, [instance])
+    check._setup_kube_client = lambda: None
+    check.kube_client = MagicMock()
+    check.kube_client.get_pods.return_value = []
+    check.kube_client.get_vms.return_value = []
+    check.kube_client.get_vmis.return_value = []
+
+    with pytest.raises(Exception):
+        dd_run_check(check)
+
+    aggregator.assert_metric(
+        "kubevirt_api.can_connect",
+        value=0,
+        tags=healthz_tags,
+    )
+
+
+def test_emits_one_can_connect_when_service_is_up(dd_run_check, aggregator, instance, mocker):
+    mocker.patch("requests.get", wraps=mock_http_responses)
+
+    check = KubeVirtApiCheck("kubevirt_api", {}, [instance])
+    check._setup_kube_client = lambda: None
+    check.kube_client = MagicMock()
+    check.kube_client.get_vms.return_value = GET_VMS_RESPONSE["items"]
+    check.kube_client.get_vmis.return_value = GET_VMIS_RESPONSE["items"]
+
+    dd_run_check(check)
+    aggregator.assert_metric(
+        "kubevirt_api.can_connect",
+        value=1,
+        tags=healthz_tags,
+    )
+
+
+BAD_METRICS_HOSTNAME_INSTANCE = {
```

Comment:
Can this and `healthz_tags` be placed in a `tests/constants.py` file?

Reply:
Done